data/lib/core/position.lua

```diff
@@ -18,7 +26,7 @@ function Position:getNextPosition(direction, steps)
 	end
 end
 
-function Position:moveUpstairs()
+function Position:moveUpstairs(cancelStep)
```

Comment:
It is probably not necessary to modify this function, when we can only compare the positions and if they are the same, then it means that no tile was found